subprojects/core-api/src/main/java/org/gradle/api/file/SourceDirectorySet.java

```diff
@@ -150,7 +150,8 @@ public interface SourceDirectorySet extends FileTree, PatternFilterable, Named,
      * @return The classes directory property for this set of sources.
      * @since 6.1
      */
-    Provider<Directory> getClassesDirectory();
+    @ReplacesEagerProperty(adapter = SourceDirectorySetAdapter.class)
+    DirectoryProperty getClassesDirectory();
```

Comment:
IIRC you mentioned you didn't know if this would be source compatible. Can you elaborate a bit? 

I think if this is not source compatible for 9.x we would still want to move forward with something like this for 10